include/nudb/impl/basic_store.ipp

```diff
@@ -217,7 +216,7 @@ insert(
     if(ecb_)
     {
         ec = ec_;
-        return false;
+        return;
     }
     // Data Record
     if(size == 0)
```

Comment:
As we discussed, undefined behavior and asserting is probably a better choice then. Don't forget to mentioned it in the docs.